torch/_dynamo/optimizations/distributed.py

```diff
@@ -39,72 +67,66 @@ def compile_fn(self, gm: fx.GraphModule, example_inputs: List[torch.Tensor]):
         """
 
         # 1: compute the partition map according to DDP bucket logic
-        bucket_bytes = 0
-        bucket_actual_sizes = []
-        node_splits = [[]]
+        buckets = [Bucket()]  # (size, param_names)
         for node in reversed(gm.graph.nodes):
-            if node.op == "output" or node.op == "placeholder":
+            if node.op in ("output", "placeholder"):
                 continue
 
-            if bucket_bytes >= self.bucket_bytes_cap:
-                bucket_actual_sizes.insert(0, bucket_bytes)
-                bucket_bytes = 0
-                node_splits.insert(0, [])
+            if (
+                buckets[0].size >= self.bucket_bytes_cap
+                or len(buckets) == 1
```

Comment:
@davidberard98 isn't the `len(buckets) == 1 and ...` good enough to make sure first bucket cap only applies to first bucket regardless of relative size to other bucket cap?